crates/core/src/repofile/snapshotfile.rs

```diff
@@ -617,19 +614,19 @@ impl SnapshotFile {
         }
         let mut latest: Option<Self> = None;
         let mut pred = predicate;
-        let mut snapshots = Vec::new();
-        for snap in be.stream_all::<Self>(p)? {
-            let (id, mut snap) = snap?;
-            if !pred(&snap) {
-                continue;
-            }
-
-            snap.id = id;
-            snapshots.push(snap);
-        }
+        let mut snapshots: Vec<_> = be
+            .stream_all::<Self>(p)?
+            .into_iter()
+            .filter_map(Result::ok)
```

Comment:
Note that this chenges the error handling: Now all errors when reading a single SnapshotFile from the backend are ignored. This is somehow desired behavior, IMO, but deserves a warning. I added one ;-)